Marlin/src/lcd/extui/anycubic_vyper/dgus_tft.cpp

```diff
@@ -1050,38 +1031,28 @@ namespace Anycubic {
 
         #if HAS_FAN
           else if (control_index == TXT_FAN_SPEED_TARGET) { // fan speed
-            p_u8 = (unsigned char *)(&control_value); // get value
-            p_u8[0] = data_buf[5];
-            p_u8[1] = data_buf[4];
-
-            temp = constrain((uint16_t)control_value, 0, 100);
-            SendValueToTFT((uint16_t)temp, TXT_FAN_SPEED_NOW);
-            SendValueToTFT((uint16_t)temp, TXT_FAN_SPEED_TARGET);
+            control_value = (uint16_t(data_buf[4]) << 8) | uint16_t(data_buf[5]);
+            temp = constrain(uint16_t(control_value), 0, 100);
+            SendValueToTFT(temp, TXT_FAN_SPEED_NOW);
+            SendValueToTFT(temp, TXT_FAN_SPEED_TARGET);
             setTargetFan_percent(temp, FAN0);
           }
         #endif
 
         else if (control_index == TXT_PRINT_SPEED_TARGET || control_index == TXT_ADJUST_SPEED) { // print speed
-          p_u8 = (unsigned char *)(&control_value); // get value
-          p_u8[0] = data_buf[5];
-          p_u8[1] = data_buf[4];
-
-          float feedrate = constrain((uint16_t)control_value, 40, 999);
+          control_value = (uint16_t(data_buf[4]) << 8) | uint16_t(data_buf[5]);
+          const uint16_t feedrate = constrain(uint16_t(control_value), 40, 999);
           //feedrate_percentage=constrain(control_value,40,999);
-          sprintf(str_buf, "%u", (uint16_t)feedrate);
+          sprintf(str_buf, "%u", feedrate);
           SendTxtToTFT(str_buf, TXT_PRINT_SPEED);
-          SendValueToTFT((uint16_t)feedrate, TXT_PRINT_SPEED_NOW);
-          SendValueToTFT((uint16_t)feedrate, TXT_PRINT_SPEED_TARGET);
+          SendValueToTFT(feedrate, TXT_PRINT_SPEED_NOW);
+          SendValueToTFT(feedrate, TXT_PRINT_SPEED_TARGET);
           setFeedrate_percent(feedrate);
         }
 
         else if (control_index == REG_LCD_READY) {
-          p_u8 = (unsigned char *)(&control_value); // get value
-          p_u8[0] = data_buf[5];
-          p_u8[1] = data_buf[4];
-          p_u8[2] = data_buf[3];
-          if ((control_value & 0x00FFFFFF) == 0x010072) { // startup last gif
-
+          control_value = (uint32_t(data_buf[3]) << 16) | (uint32_t(data_buf[4]) << 8) | uint32_t(data_buf[5]);
+          if (control_value == 0x010072) { // startup last gif
```

Comment:
@Bob-the-Kuhn — The `control_value & 0x00FFFFFF` is not needed here, however, since `control_value` is initialized to 0 before this, and it is not otherwise modified.